File: back-end/src/main/java/com/example/backend/services/AuthService.java
package com.example.backend.services;

import com.example.backend.dto.LoginRequest;
import com.example.backend.dto.LoginResponse;
import com.example.backend.dto.SchoolRequest;
import com.example.backend.dto.SchoolResponse;

public interface AuthService {
    SchoolResponse register(SchoolRequest schoolRequest);
    LoginResponse login(LoginRequest loginRequest);
    void logout();
}

File: front-end/src/services/AuthService.ts
import { AuthResponse, RegisterRequest } from '../core/dto/auth.dto';
import { User } from "../core/entities/user.entity";

export interface AuthService {
  isAuthenticated(): boolean;
  login(username: string, password: string): Promise<void>;
  logout(): Promise<void>;
  register(data: RegisterRequest): Promise<AuthResponse>;
  getCurrentUser(): Promise<User>;
  hasRole(role: string): boolean;
  getToken(): string;
}

